python/GafferSceneUI/PrimitiveVariableTweaksUI.py

```diff
@@ -0,0 +1,383 @@
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import imath
+import functools
+import collections
+
+import IECore
+import IECoreScene
+
+import Gaffer
+import GafferUI
+import GafferScene
+import GafferSceneUI
+
+Gaffer.Metadata.registerNode(
+
+	GafferScene.PrimitiveVariableTweaks,
+
+	"description",
+	"""
+	Modify primitive variable values. Supports modifying values just for specific elements of the
+	primitive.
+	""",
+
+	"layout:activator:selectionModeEnabled", lambda node : node["interpolation"].getValue() != IECoreScene.PrimitiveVariable.Interpolation.Invalid,
+	"layout:activator:idListExplicitVisible", lambda node : node["interpolation"].getValue() != IECoreScene.PrimitiveVariable.Interpolation.Invalid and node["selectionMode"].getValue() == GafferScene.PrimitiveVariableTweaks.SelectionMode.IdList,
+	"layout:activator:idListVarVisible", lambda node : node["interpolation"].getValue() != IECoreScene.PrimitiveVariable.Interpolation.Invalid and node["selectionMode"].getValue() == GafferScene.PrimitiveVariableTweaks.SelectionMode.IdListPrimitiveVariable,
+	"layout:activator:idListVisible", lambda node : node["interpolation"].getValue() != IECoreScene.PrimitiveVariable.Interpolation.Invalid and node["selectionMode"].getValue() in [ GafferScene.PrimitiveVariableTweaks.SelectionMode.IdList, GafferScene.PrimitiveVariableTweaks.SelectionMode.IdListPrimitiveVariable ],
+	"layout:activator:maskVarVisible", lambda node : node["interpolation"].getValue() != IECoreScene.PrimitiveVariable.Interpolation.Invalid and node["selectionMode"].getValue() == GafferScene.PrimitiveVariableTweaks.SelectionMode.MaskPrimitiveVariable,
+	plugs = {
+
+		"interpolation" : [
+
+			"description",
+			"""
+			The interpolation of the target primitive variables. Using "Any" allows you to
+			operate on any primitive variable, but if you know your target, using a more
+			specific interpolation offers benefits: you can specify an idList to operate
+			on specific elements, and you can use "Create" mode to create new primitive
+			variables.
+			""",
+
+			"preset:Any", IECoreScene.PrimitiveVariable.Interpolation.Invalid,
+			"preset:Constant", IECoreScene.PrimitiveVariable.Interpolation.Constant,
+			"preset:Uniform", IECoreScene.PrimitiveVariable.Interpolation.Uniform,
+			"preset:Vertex", IECoreScene.PrimitiveVariable.Interpolation.Vertex,
+			"preset:Varying", IECoreScene.PrimitiveVariable.Interpolation.Varying,
+			"preset:FaceVarying", IECoreScene.PrimitiveVariable.Interpolation.FaceVarying,
+
+			"plugValueWidget:type", "GafferUI.PresetsPlugValueWidget",
+
+		],
+
+		"selectionMode" : [
+
+			"description",
+			"""
+			Choose how to select which elements are affected. Only takes effect if you
```

Comment:
Oh, I was initially confused by this ... looks like I omitted my final round of changes from this PR. Fortunately, the only things missing were some minor formatting fixes, and a UI fixup where I did update the UI to reflect the current behaviour.

I'm open to discussing changing this behaviour, but it would make everything a bit more complicated. Currently the rule is: "When tweaking a Constant primvar, use a tweak that matches the type of the primvar. When tweaking any other interpolation, use a tweak that match the type of one element of the primvar." If we want to support per-element tweaks of Constant vector primvars ( I assume this would include both using an IdList to set certain elements, or something like using a Color3f plug to add a value to every element of a Color3fVector constant ), that would be a new code path. Currently, you can only apply ListAppend, ListPrepend, ListRemove value tweaks to vector Constants. Do you have a use case in mind to warrant the extra complexity?